### ISSUE TITLE
Validation error when specifying Element Attributes

### ISSUE THREAD

**Issue by BwehaaFox** [typed-ember/glint#738]
After installing glint in our project, generally valid vscode works well, except for one thing. If I import a component and try to give it, for example, a `class` attribute, it is flagged as an error:

```
Argument of type 'unknown' is not assignable to parameter of type 'Element'.glint(2345)
```

This works with any attribute without `@` on using `...attributes`
I saw in examples that there should be no errors with this, so I don’t understand where it comes from

![](https://e.radikal.host/2024/06/25/imagecb632e8266152bdf.png)

If you disable the vscode plugin, there is no error, no validation, so I assumed it was related to this addon

```json
"@glint/core": "^1.4.0",
"@glint/environment-ember-loose": "^1.4.0",
"@glint/environment-ember-template-imports": "^1.4.0",
"@glint/template": "^1.4.0",
"ember-template-imports": "^4.1.1",
```

``` json
{
  "compilerOptions": {
    "target": "ESNext",
    "lib": [
      "es2016",
      "dom"
    ],
    "allowJs": true,
    "allowSyntheticDefaultImports": true,
    "noImplicitAny": false,
    "noImplicitThis": true,
    "alwaysStrict": true,
    "strictNullChecks": true,
    "strictPropertyInitialization": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "noEmitOnError": false,
    "noEmit": true,
    "inlineSourceMap": true,
    "inlineSources": true,
    "baseUrl": ".",
    "module": "NodeNext",
    "moduleResolution": "NodeNext",
    "experimentalDecorators": true,
    "types": ["ember-source/types"],
    "paths": {
      "*": [
        "types/*"
      ],
      "@glimmer/tracking": [
        "node_modules/ember-cached-decorator-polyfill",
        "node_modules/@glimmer/tracking/dist/types"
      ],
    }
  },
  "exclude": ["node_modules", "dist", "lib"],
  "include": [
    "app/**/*",
    "tests/**/*",
    "types/**/*"
  ],
  "glint": {
    "environment": [
      "ember-loose",
      "ember-template-imports"
    ]
  }
}
```

**Comment by NullVoxPopuli** [comment]
The signature for ui container probably needs to be an HTMLElement, rather than just Element.

**Comment by BwehaaFox** [comment]
То есть для компонентов старого формата вместо опционального Args интерфейса вроде:

```ts
interface Args {
  name: string;
}

export default class extends Component<Args>
```

Нужно декларативно указывать сигнатуру вроде этой для корректной работы валидации?

```ts
interface Signature {
  Args: {
    name: string;
  };
  Element: HTMLElement;
}

export default class extends Component<Signature>
```

**Comment by NullVoxPopuli** [comment]
Yes, you need the full signature, as in your second code snippet.


Also, this Android feature is amazing!

![Screenshot_20240625-072817](https://github.com/typed-ember/glint/assets/199018/09e66e0a-e89c-440c-90df-9ef1464f67d9)

**Comment by BwehaaFox** [comment]
Sorry, I forgot to translate the text in advance 😅. English is not at a sufficient level to formulate the primary thought in it, I’ll correct it now

I agree, built-in translation is indispensable

**Comment by NullVoxPopuli** [comment]
no worries!! I meant for the other non-russian-speaking folks -- glad you're problem is solved!
